WordPress/Classes/ViewRelated/Reader/ReaderPostDetailViewController.m

```diff
@@ -265,6 +265,32 @@ - (void)setupWithPostID:(NSNumber *)postID siteID:(NSNumber *)siteID
 
 #pragma mark - Accessor methods
 
+- (void)setPost:(ReaderPost *)post
+{
+    if (post == _post) {
+        return;
+    }
+
+    if (!post) {
+        _post = nil;
+        return;
+    }
+
+    _post = [self postInContext:post.objectID];
```

Comment:
(Tested, and it's working fine!). I was thinking... would it be possible for `ReaderPostsViewController` not to use a derived context?

Reason: [this other spot](https://github.com/wordpress-mobile/WordPress-iOS/blob/issues/4292-like-button/WordPress/Classes/ViewRelated/Reader/ReaderPostsViewController.m#L1162) would require the exact same patch (and it'd be great if we could solve it all in one shot).